dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+		VelocityEngine ve = VelocityUtil.getEngine();
+		StringWriter sw = new StringWriter();
+		ve.evaluate(ctx, sw, "velocity eval", velocity);
+		return sw.toString();
+		
+	}
+	private static ToolboxManager toolboxManager=null;
+    public static ToolboxManager getToolboxManager () {
+        if ( toolboxManager == null ) {
+            synchronized ( VelocityUtil.class ) {
+                if ( toolboxManager == null ) {
+                    toolboxManager = ServletToolboxManager.getInstance( Config.CONTEXT, Config.getStringProperty("TOOLBOX_MANAGER_PATH", "/WEB-INF/toolbox.xml"));
+                }
+            }
+
+        }
+        return toolboxManager;
+    }
+	
+	private static List<Language> languages =null;
+	
+	private static List<Language> getLanguages(){
+		if(languages ==null){
+			synchronized (VelocityUtil.class) {
+				if(languages ==null){
+					languages = APILocator.getLanguageAPI().getLanguages();
+				}
+			}
+		}
+		return languages;
+		
+	}
+	
+
+	   public static void makeBackendContext(Context context, IHTMLPage htmlPage, String cmsTemplateInode, String idURI, HttpServletRequest request,
+	           PageMode mode, Host host) throws DotDataException {
+	
+	       context.put("context", context);
+
+	        // stick some useful variables in the context
+	        if (htmlPage != null) {
+	            context.put("HTMLPAGE_INODE", String.valueOf(htmlPage.getInode()));
+	            context.put("HTMLPAGE_IDENTIFIER", String.valueOf(htmlPage.getIdentifier()));
+	            context.put("HTMLPAGE_TITLE", htmlPage.getTitle());
+	            context.put("HTMLPAGE_META", htmlPage.getMetadata());
+	            //http://jira.dotmarketing.net/browse/DOTCMS-6427
+	            context.put("HTMLPAGE_DESCRIPTION", htmlPage.getSeoDescription());
+	            context.put("HTMLPAGE_KEYWORDS", htmlPage.getSeoKeywords());
+	            context.put("HTMLPAGE_SECURE", String.valueOf(htmlPage.isHttpsRequired()));
+	            context.put("HTMLPAGE_REDIRECT", htmlPage.getRedirect());
+	            context.put("friendlyName", htmlPage.getFriendlyName());
+	            context.put("pageTitle", htmlPage.getTitle());
+                context.put("dotPageMode", mode);
+	            Date moddate = htmlPage.getModDate();
+
+	            moddate = new Date(moddate.getTime());
+
+	            context.put("HTML_PAGE_LAST_MOD_DATE", moddate);
+
+	            try {
+	                context.put("htmlPageInode", htmlPage.getInode());
+
+	                // for browsing the tree
+	                String view = java.net.URLEncoder.encode("(working=" + com.dotmarketing.db.DbConnectionFactory.getDBTrue()
+	                        + " and deleted=" + com.dotmarketing.db.DbConnectionFactory.getDBFalse() + "and language_id = "
+	                        + (String) request.getSession().getAttribute(com.dotmarketing.util.WebKeys.HTMLPAGE_LANGUAGE) + ")", "UTF-8");
+	                context.put("view", view);
+	            } catch (Exception e) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Catch a list of specific exception subtypes instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2221)